prusa/link/multi_instance/__main__.py

```diff
@@ -42,6 +42,30 @@ def get_logger_file_descriptors():
     return file_descriptors
 
 
+class StderrLogAdapter:
+    """This class is used to redirect stderr to a logger"""
+
+    def __init__(self, name, level=logging.INFO):
+        self.logger = logging.getLogger(name)
+        self.level = level
+        self.buffer = []
+
+    def write(self, message):
```

Comment:
Why to complex?  And you miss some functionality (fileno). See commit 36a72be5a4fa07253c08b60fdd61f4d7ae282d8d, there was stderr adapters yet

Reply:
![image](https://github.com/prusa3d/Prusa-Link/assets/5272796/2d5c94ec-a333-4236-8635-1068f550802f)


I do not need to support fileno() so I don't. Adding this extra useless bit of code would not improve anything.

```

def fileno(self):
    if len(self.logger.handlers) > 1:
        raise RuntimeError("Cannot retrieve file descriptor when multiple "
                           "handlers are present")
    if not self.logger.handlers:
        raise RuntimeError("Cannot retrieve file descriptor when no "
                           "handlers are present")
    handler = self.logger.handlers[0]
    if hasattr(handler, "stream"):
        return handler.stream.fileno()
    if hasattr(handler, "socket"):
        return handler.socket.fileno()
    raise RuntimeError("The log handler has no file descriptor")
```

Before, i tried calling log every write(). It looked like this:
```
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: Traceback (most recent call last):
                                                  {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:  {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   File "/usr/bin/prusalink-manager", line 8, in <module>
                                                  {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:  {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:      {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: sys.exit(main()) {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: 
                                                  {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:  {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR:   {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: ^ {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: ^ {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: ^ {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: ^ {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: ^ {write():52}
Jun 12 21:52:41 TojikG5 mi_server_stderr[34862]: ERROR: ^ {write():52}
```

Buffering made it look like this
```
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR: Traceback (most recent call last): {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:   File "/usr/bin/prusalink-manager", line 8, in <module> {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:     sys.exit(main()) {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:              ^^^^^^ {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:   File "/usr/lib/python3.11/site-packages/prusa/link/multi_instance/__main__.py", line 278, in main {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:     start(user_info) {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:   File "/usr/lib/python3.11/site-packages/prusa/link/multi_instance/__main__.py", line 166, in start {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:     Server(user_info) {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:   File "/usr/lib/python3.11/site-packages/prusa/link/multi_instance/__main__.py", line 126, in __init__ {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:     with context: {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:   File "/usr/lib/python3.11/site-packages/daemon/daemon.py", line 365, in __enter__ {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:     self.open() {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:   File "/usr/lib/python3.11/site-packages/daemon/daemon.py", line 354, in open {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:     redirect_stream(sys.stderr, self.stderr) {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:   File "/usr/lib/python3.11/site-packages/daemon/daemon.py", line 980, in redirect_stream {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:     os.dup2(target_fd, system_stream.fileno()) {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:                        ^^^^^^^^^^^^^^^^^^^^ {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR: AttributeError: 'StderrLogAdapter' object has no attribute 'fileno' {flush():63}
Jun 12 22:12:55 TojikG5 mi_server_stderr[36693]: ERROR:  {flush():63}
```